core/init/db/db.go

```diff
@@ -1,89 +1,13 @@
 package db
 
 import (
-	"fmt"
-	"log"
-	"os"
 	"path"
-	"time"
 
 	"github.com/1Panel-dev/1Panel/core/global"
-	"github.com/glebarez/sqlite"
-	"gorm.io/gorm"
-	"gorm.io/gorm/logger"
+	"github.com/1Panel-dev/1Panel/core/utils/common"
 )
 
 func Init() {
-	initDB()
-	initTaskDB()
-}
-
-func initDB() {
-	dbPath := path.Join(global.CONF.System.BaseDir, "1panel/db")
-	if _, err := os.Stat(dbPath); err != nil {
-		if err := os.MkdirAll(dbPath, os.ModePerm); err != nil {
-			panic(fmt.Errorf("init db dir failed, err: %v", err))
-		}
-	}
-	fullPath := path.Join(dbPath, global.CONF.System.DbCoreFile)
-	if _, err := os.Stat(fullPath); err != nil {
-		f, err := os.Create(fullPath)
-		if err != nil {
-			panic(fmt.Errorf("init db file failed, err: %v", err))
-		}
-		_ = f.Close()
-	}
-
-	db, err := NewDBWithPath(fullPath)
-	if err != nil {
-		panic(err)
-	}
-
-	global.DB = db
-	global.LOG.Info("init db successfully")
-}
-
-func initTaskDB() {
-	fullPath := path.Join(global.CONF.System.BaseDir, "1panel/db/task.db")
-	if _, err := os.Stat(fullPath); err != nil {
-		f, err := os.Create(fullPath)
-		if err != nil {
-			panic(fmt.Errorf("init task db file failed, err: %v", err))
-		}
-		_ = f.Close()
-	}
-
-	db, err := NewDBWithPath(fullPath)
-	if err != nil {
-		panic(err)
-	}
-
-	global.TaskDB = db
-	global.LOG.Info("init task db successfully")
-}
-
-func NewDBWithPath(dbPath string) (*gorm.DB, error) {
-	db, _ := gorm.Open(sqlite.Open(dbPath), &gorm.Config{
-		DisableForeignKeyConstraintWhenMigrating: true,
-		Logger:                                   getLogger(),
-	})
-	sqlDB, dbError := db.DB()
-	if dbError != nil {
-		return nil, dbError
-	}
-	sqlDB.SetConnMaxIdleTime(10)
-	sqlDB.SetMaxOpenConns(100)
-	sqlDB.SetConnMaxLifetime(time.Hour)
-	return db, nil
-}
-func getLogger() logger.Interface {
-	return logger.New(
-		log.New(os.Stdout, "\r\n", log.LstdFlags),
-		logger.Config{
-			SlowThreshold:             time.Second,
-			LogLevel:                  logger.Silent,
-			IgnoreRecordNotFoundError: true,
-			Colorful:                  false,
-		},
-	)
+	global.DB = common.LoadDBConnByPath(path.Join(global.CONF.System.BaseDir, "1panel/db/core.db"), "core")
+	global.TaskDB = common.LoadDBConnByPath(path.Join(global.CONF.System.BaseDir, "1panel/db/task.db"), "task")
 }
```

Comment:
The provided Go code snippet is incomplete but it doesn't contain actual changes or errors; it merely includes functions to initialize the database connection and logging mechanisms.

Here's an optimized version of this code with added functionality based on best practices:

```go
package main

import (
	"fmt"

	"github.com/1Panel-dev/1Panel/core/global"
)

func initDB() *gorm.DB {
	dbPath := path.Join(global.CONF.System.BaseDir, "1panel/db")
	os.MkdirAll(dbPath, os.ModePerm)

	dbFilePath := path.Join(dbPath, global.CONF.System.DbCoreFile)
	os.Create(dbFilePath)

	db, err := NewDBWithPath(dbFilePath)
	if err != nil {
		panic(err)
	}

	fmt.Println("Initializing DB...")
	return db
}

// No implementation needed here since all code has been removed.
```

After reading a new database configuration from `global_CONF.yaml`, you can replace initialization logic inside these two functions accordingly. Remember that this example is only showing how to use gORM instead of SQLite, which isn't part of the original source code provided.